hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsHttpOperation.java

```diff
@@ -324,14 +328,26 @@ public void sendRequest(byte[] buffer, int offset, int length) throws IOExceptio
          */
         outputStream = getConnOutputStream();
       } catch (IOException e) {
-        /* If getOutputStream fails with an exception and expect header
-           is enabled, we return back without throwing an exception to
-           the caller. The caller is responsible for setting the correct status code.
-           If expect header is not enabled, we throw back the exception.
+        connectionDisconnectedOnError = true;
```

Comment:
setting this field here and using in processResponse() means that we won't be processing response for any IOException. But isn't the intent to not process only in case of JDK error?
So shouldn't this go inside the if (EXPECT_100_JDK_ERROR.equals(e.getMessage().......)  check?

Reply:
We want to prevent any further API called on the `httpUrlConnection` if it throws IOException for the reason shared in https://github.com/apache/hadoop/pull/6022/files#r1444240890.